general-concepts/dependencies/text.xml

```diff
@@ -850,5 +850,309 @@ USE flags. This would then break building your ebuild.
 
 </body>
 </section>
+
+<section>
+<title>Common pitfalls</title>
+<body>
+
+<p>
+The following pitfalls occur frequently when dealing with slots, slot
+operators, any-of dependencies, and blockers. They are easy to miss and can lead
+to subtle resolver behaviour or rebuild issues. The guidance below is a brief
+summary; see the linked references for deeper background.
+</p>
+
+</body>
+
+<subsection>
+<title>Separate dependency atoms are not combined into one slot</title>
+<body>
+
+<p>
+Multiple independent atoms are not guaranteed to be satisfied by the same
+slot of a slotted package. For example, a common version range like the
+following can be satisfied by two different slots:
+</p>
+
+<codesample lang="ebuild">
+# Bad (if app-misc/foo is slotted):
+DEPEND="&gt;=app-misc/foo-2
+	&lt;app-misc/foo-5"
+</codesample>
+
+<p>
+If <c>app-misc/foo</c> is slotted and two installed slots each satisfy one
+side of the range, the resolver can legally use them separately. Likewise,
+independent USE requirements:
+</p>
+
+<codesample lang="ebuild">
+# Bad (if app-misc/foo is slotted):
+RDEPEND="app-misc/foo[foo]
+	bar? ( app-misc/foo[baz] )"
+</codesample>
+
+<p>
+may be met by different slots with different USE configurations. For
+packages that are not truly multi-slotted (i.e. slots represent mutually
+incompatible ABIs), the safe fix is to state the intended slot explicitly or
+to express the intent via any-of groups limited per slot.
```

Comment:
Maybe provide examples for both proposed solutions? The examples below correspond to other cases, right?